src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,72 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  function handleCloseModal() {
+    dispatch(currentTodoSlice.actions.changeCurrentTodo(null));
+  }
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(currentTodo!.userId)
```

Comment:
The use of `currentTodo!` assumes that `currentTodo` is always defined. Ensure that `currentTodo` is not null or undefined before accessing its properties to avoid runtime errors.